### PR TITLE
MAINT: Remove Python2 specific C module setup [part2]

### PR DESCRIPTION
**Pull request by sethtroisi**
Some more python2 cleanups building on https://github.com/numpy/numpy/pull/14590

### CODE REVIEW

File: numpy/f2py/rules.py [hunk 1]
```diff
@@ -211,7 +205,7 @@
 \tPy_TYPE(&PyFortran_Type) = &PyType_Type;
 \timport_array();
 \tif (PyErr_Occurred())
-\t\t{PyErr_SetString(PyExc_ImportError, \"can't initialize module #modulename# (failed to import numpy)\"); return RETVAL;}
```
Comment (eric-wieser) [inline numpy/f2py/rules.py]: This looks like it should be `return NULL`, but let's leave that to a follow-up

**Review by eric-wieser — APPROVED** [review]
Looks great, thanks!

**Comment by seberg** [comment]
LGTM as well, I would be for simply squash merging this one and closing the previous one here.

**Comment by eric-wieser** [comment]
CI failure looks unrelated

**Comment by seberg** [comment]
Thanks!

File: numpy/f2py/rules.py (hunk 1 above)
Comment (sethtroisi) [reply numpy/f2py/rules.py]: this pattern repeats in a number of places I have a draft pull to try something new..

**Pull request merged**